specification/databasewatcher/DatabaseWatcher.Management/alertRuleResources.tsp

```diff
@@ -0,0 +1,54 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+namespace Microsoft.DatabaseWatcher;
+
+using Azure.ResourceManager;
+using TypeSpec.Versioning;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+
+@armResourceOperations
+interface AlertRuleResources {
+  get is ArmResourceRead<AlertRuleResource>;
+  createOrUpdate is ArmResourceCreateOrReplaceSync<AlertRuleResource>;
+  delete is ArmResourceDeleteSync<AlertRuleResource>;
+  listByParent is ArmResourceListByParent<AlertRuleResource>;
+}
+
+@parentResource(Watcher)
+model AlertRuleResource is ProxyResource<AlertRuleResourceProperties> {
+  @doc("The alert rule proxy resource name.")
+  @key("alertRuleResourceName")
+  @pattern("^[a-zA-Z0-9]{1}[a-zA-Z0-9-_.]{0,62}[a-zA-Z0-9_]{1}$")
+  @segment("alertRuleResources")
+  @path
+  name: string;
+}
+
+@doc("The generic properties of the alert rule proxy resource.")
+model AlertRuleResourceProperties {
+  @doc("The resource ID of the alert rule resource.")
+  alertRuleResourceId: Azure.Core.armResourceIdentifier<[
+    {
+      type: "Microsoft.Insights/scheduledQueryRules";
+    }
+  ]>;
+
+  @doc("Whether the alert rule was created with an action group.")
+  createdWithActionGroup: boolean;
```

Comment:
>createdWithActionGroup [](http://example.com/codeflow?start=2&length=22)

you can convert this boolean property to an enum so that the model can be more extensible if the an alert rule could be created with other types of features . Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way

Reply:
Was this comment addressed?